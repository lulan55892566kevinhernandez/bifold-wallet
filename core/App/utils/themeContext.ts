import { createContext, useContext } from 'react'

import { defaultTheme, Theme } from '../theme'

export { defaultTheme } from '../theme'
export type { Theme } from '../theme'
export const ThemeContext = createContext<Theme>(defaultTheme)
export const ThemeProvider = ThemeContext.Provider

export const useThemeContext = () => useContext(ThemeContext)
