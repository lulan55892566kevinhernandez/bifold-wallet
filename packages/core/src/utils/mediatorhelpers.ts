import { Agent, MediationRecord } from '@credo-ts/core'

export const isMediatorInvitation = async (agent: Agent, url: string): Promise<boolean> => {
  const invitation = await agent.oob.parseInvitation(url)
  if (!invitation) {
    return false
  }
  if (invitation.goalCode === 'aries.vc.mediate') {
    agent.config.logger.info(`Invitation is a mediator invitation with goal code: ${invitation.goalCode}`)
    return true
  }
  agent.config.logger.info(`Invitation is not a mediator invitation, goal code: ${invitation.goalCode}`)
  return false
}

const provisionMediationRecordFromMediatorUrl = async (
  agent: Agent,
  url: string
): Promise<MediationRecord | undefined> => {
  try {
    const invitation = await agent.oob.parseInvitation(url)
    if (!invitation) {
      agent.config.logger.warn(`No invitation found in URL: ${url}`)
      return undefined
    }
    const outOfBandRecord = await agent.oob.findByReceivedInvitationId(invitation.id)
    let [connection] = outOfBandRecord ? await agent.connections.findAllByOutOfBandId(outOfBandRecord.id) : []

    if (!connection) {
      agent.config.logger.warn(`No connection found for out-of-band record: ${outOfBandRecord?.id}`)
      const invite = await agent.oob.parseInvitation(url)
      const { connectionRecord: newConnection } = await agent.oob.receiveInvitation(invite)
      if (!newConnection) {
        agent.config.logger.warn(`Failed to create connection from invitation: ${JSON.stringify(invite, null, 2)}`)
        return
      }
      connection = newConnection
    }
    const result = connection.isReady ? connection : await agent.connections.returnWhenIsConnected(connection.id)
    return agent.mediationRecipient.provision(result)
  } catch (error) {
    agent.config.logger.warn(`Failed to get connection ID from mediator URL: ${error}`)
    return
  }
}

export const setMediationToDefault = async (agent: Agent, mediatorUrl: string) => {
  const mediationRecord = await provisionMediationRecordFromMediatorUrl(agent, mediatorUrl)
  if (!mediationRecord) {
    agent.config.logger.warn(`No connection record found for mediator URL: ${mediatorUrl}`)
    return
  }
  const currentDefault = await agent.mediationRecipient.findDefaultMediator()
  if (currentDefault?.connectionId === mediationRecord.id) {
    agent.config.logger.info(`Default mediator already set for connection ID: ${mediationRecord.id}`)
    return
  }
  await agent.mediationRecipient.setDefaultMediator(mediationRecord)
  agent.config.logger.info(`setting default mediator with record: ${JSON.stringify(mediationRecord)}`)
}
