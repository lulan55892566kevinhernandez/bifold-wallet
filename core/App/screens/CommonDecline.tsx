import { CredentialState, ProofState } from '@aries-framework/core'
import { useAgent, useCredentialById, useProofById } from '@aries-framework/react-hooks'
import { StackScreenProps } from '@react-navigation/stack'
import React, { useState, useEffect } from 'react'
import { useTranslation } from 'react-i18next'
import { ScrollView, StatusBar, StyleSheet, Text, View } from 'react-native'
import { Edge, SafeAreaView } from 'react-native-safe-area-context'

import CredentialDeclined from '../assets/img/credential-declined.svg'
import ProofRequestDeclined from '../assets/img/proof-declined.svg'
import Button, { ButtonType } from '../components/buttons/Button'
import CredentialCard from '../components/misc/CredentialCard'
import InfoBox, { InfoBoxType } from '../components/misc/InfoBox'
import { DispatchAction } from '../contexts/reducers/store'
import { useStore } from '../contexts/store'
import { useTheme } from '../contexts/theme'
import { DeclineType } from '../types/decline'
import { BifoldError } from '../types/error'
import { TabStacks, NotificationStackParams, Screens } from '../types/navigators'
import { testIdWithKey } from '../utils/testable'

type CommonDeclineProps = StackScreenProps<NotificationStackParams, Screens.CommonDecline>

const CommonDecline: React.FC<CommonDeclineProps> = ({ navigation, route }) => {
  if (!route?.params) {
    throw new Error('CommonDecline route prams were not set properly')
  }

  const { declineType, itemId } = route.params

  if (!itemId) {
    throw new Error('itemId cannot be undefined')
  }

  if (!declineType) {
    throw new Error('declineType cannot be undefined')
  }

  const { agent } = useAgent()
  const credential = useCredentialById(itemId)
  const proof = useProofById(itemId)
  const [, dispatch] = useStore()
  const { t } = useTranslation()
  const { ColorPallet, TextTheme } = useTheme()
  const [didDecline, setDidDecline] = useState<boolean>(false)
  const imageDisplayOptions = {
    fill: ColorPallet.notification.infoText,
    height: 250,
    width: 250,
  }
  const styles = StyleSheet.create({
    container: {
      height: '100%',
      backgroundColor: ColorPallet.brand.primaryBackground,
      padding: 20,
    },
    image: {
      marginVertical: 66,
    },
    messageContainer: {
      marginTop: 25,
      alignItems: 'center',
    },
    messageText: {
      fontWeight: 'normal',
      textAlign: 'center',
      marginTop: 30,
    },
  })

  if (
    (declineType === DeclineType.ProofRequest && credential) ||
    (declineType === DeclineType.CredentialOffer && proof)
  ) {
    throw new Error('Usage type and artifact do not match')
  }

  const onDoneTouched = () => {
    navigation.getParent()?.navigate(TabStacks.HomeStack, { screen: Screens.Home })
  }

  const confirmDeclineTouched = async () => {
    try {
      if (agent && credential) {
        await agent.credentials.declineOffer(credential.id)
        return
      }

      if (agent && proof) {
        await agent.proofs.declineRequest(proof.id)
        return
      }
    } catch (err: unknown) {
      const error = new BifoldError(
        declineType === DeclineType.ProofRequest ? t('Error.Title1028') : t('Error.Title1025'),
        declineType === DeclineType.ProofRequest ? t('Error.Message1028') : t('Error.Message1025'),
        (err as Error).message,
        1025
      )

      dispatch({
        type: DispatchAction.ERROR_ADDED,
        payload: [{ error }],
      })
    }
  }

  useEffect(() => {
    if (credential) {
      navigation.setOptions({ title: t('CredentialOffer.DeclineTitle') })
    }

    if (credential && credential.state === CredentialState.Declined) {
      navigation.setOptions({ headerShown: false })
      setDidDecline(true)
    }
  }, [credential])

  useEffect(() => {
    if (proof) {
      navigation.setOptions({ title: t('ProofRequest.DeclineTitle') })
    }

    if (proof && proof.state === ProofState.Declined) {
      navigation.setOptions({ headerShown: false })
      setDidDecline(true)
    }
  }, [proof])

  return (
    <SafeAreaView edges={['left', 'right', 'bottom'].concat(didDecline ? ['top'] : []) as Edge[]}>
      {didDecline && <StatusBar hidden={true} />}
      <>
        {!didDecline && (
          <>
            <ScrollView style={[styles.container]}>
              <InfoBox
                notificationType={InfoBoxType.Warn}
                title={
                  declineType === DeclineType.ProofRequest
                    ? t('ProofRequest.ConfirmDeclinedTitle')
                    : t('CredentialOffer.ConfirmDeclinedTitle')
                }
                description={
                  declineType === DeclineType.ProofRequest
                    ? t('ProofRequest.ConfirmDeclinedMessage')
                    : t('CredentialOffer.ConfirmDeclinedMessage')
                }
              />
              {credential && (
                <View style={{ marginTop: 20 }}>
                  <CredentialCard credential={credential} />
                </View>
              )}
            </ScrollView>

            <View style={{ marginTop: 'auto', margin: 20 }}>
              <View style={{ paddingTop: 10 }}>
                <Button
                  title={
                    declineType === DeclineType.ProofRequest
                      ? t('ProofRequest.ConfirmDecline')
                      : t('CredentialOffer.ConfirmDecline')
                  }
                  accessibilityLabel={
                    declineType === DeclineType.ProofRequest
                      ? t('ProofRequest.ConfirmDecline')
                      : t('CredentialOffer.ConfirmDecline')
                  }
                  testID={testIdWithKey('ConfirmDeclineButton')}
                  onPress={confirmDeclineTouched}
                  buttonType={ButtonType.Primary}
                />
              </View>
              <View style={[{ paddingTop: 10 }]}>
                <Button
                  title={
                    declineType === DeclineType.ProofRequest
                      ? t('ProofRequest.AbortDecline')
                      : t('CredentialOffer.AbortDecline')
                  }
                  accessibilityLabel={
                    declineType === DeclineType.ProofRequest
                      ? t('ProofRequest.AbortDecline')
                      : t('CredentialOffer.AbortDecline')
                  }
                  testID={testIdWithKey('AbortDeclineButton')}
                  onPress={() => {
                    navigation.pop()
                  }}
                  buttonType={ButtonType.Secondary}
                />
              </View>
            </View>
          </>
        )}

        {didDecline && (
          <>
            <ScrollView style={[styles.container]}>
              <View style={[styles.messageContainer]}>
                <Text
                  style={[TextTheme.headingThree, styles.messageText]}
                  testID={testIdWithKey('RequestOrOfferDeclined')}
                >
                  {declineType === DeclineType.ProofRequest
                    ? t('ProofRequest.ProofRequestDeclined')
                    : t('CredentialOffer.CredentialDeclined')}
                </Text>
                {declineType === DeclineType.ProofRequest ? (
                  <ProofRequestDeclined style={[styles.image]} {...imageDisplayOptions} />
                ) : (
                  <CredentialDeclined style={[styles.image]} {...imageDisplayOptions} />
                )}
              </View>
            </ScrollView>

            <View style={{ marginTop: 'auto', margin: 20 }}>
              <Button
                title={t('Global.Done')}
                accessibilityLabel={t('Global.Done')}
                testID={testIdWithKey('Done')}
                onPress={onDoneTouched}
                buttonType={ButtonType.Primary}
              />
            </View>
          </>
        )}
      </>
    </SafeAreaView>
  )
}

export default CommonDecline
