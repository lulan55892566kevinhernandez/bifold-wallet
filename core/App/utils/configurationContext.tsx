import { createContext, useContext } from 'react'

import { pages } from '../screens/OnboardingPages'

export interface ConfigurationContext {
  onboarding: {
    pages: typeof pages
  }
  splash: React.FC
  terms: React.FC
}

const ConfigurationContext = createContext<ConfigurationContext>(null as unknown as ConfigurationContext)
export const ConfigurationProvider = ConfigurationContext.Provider

export const useConfigurationContext = () => useContext(ConfigurationContext)
