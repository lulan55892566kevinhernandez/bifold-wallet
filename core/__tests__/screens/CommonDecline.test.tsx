import { render, fireEvent } from '@testing-library/react-native'

import React from 'react'

import CommonDecline, { DeclineType } from '../../App/screens/CommonDecline'
import { testIdWithKey } from '../../App/utils/testable'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')
jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})

const onGoBackTouched = jest.fn()
const onDeclinedConformationTouched = jest.fn()

describe('common decline screen', () => {
  beforeEach(() => {
    onGoBackTouched.mockReset()
    onDeclinedConformationTouched.mockReset()
  })

  test('decline proof renders correctly', () => {
    const tree = render(
      <CommonDecline
        visible={true}
        declineType={DeclineType.ProofRequest}
        didDeclineOfferOrProof={false}
        onGoBackTouched={onGoBackTouched}
        onDeclinedConformationTouched={onDeclinedConformationTouched}
      />
    )

    const confirmDeclineButton = tree.getByTestId(testIdWithKey('ConfirmDeclineButton'))
    const abortDeclineButton = tree.getByTestId(testIdWithKey('AbortDeclineButton'))

    expect(tree).toMatchSnapshot()
    expect(confirmDeclineButton).not.toBeNull()
    expect(abortDeclineButton).not.toBeNull()
  })

  test('user confirms decline', () => {
    const tree = render(
      <CommonDecline
        visible={true}
        declineType={DeclineType.ProofRequest}
        didDeclineOfferOrProof={false}
        onGoBackTouched={onGoBackTouched}
        onDeclinedConformationTouched={onDeclinedConformationTouched}
      />
    )

    const confirmDeclineButton = tree.getByTestId(testIdWithKey('ConfirmDeclineButton'))

    fireEvent(confirmDeclineButton, 'press')

    expect(onGoBackTouched).toBeCalledTimes(0)
    expect(onDeclinedConformationTouched).toBeCalledTimes(1)
  })

  test('user aborts decline', () => {
    const tree = render(
      <CommonDecline
        visible={true}
        declineType={DeclineType.ProofRequest}
        didDeclineOfferOrProof={false}
        onGoBackTouched={onGoBackTouched}
        onDeclinedConformationTouched={onDeclinedConformationTouched}
      />
    )

    const abortDeclineButton = tree.getByTestId(testIdWithKey('AbortDeclineButton'))

    fireEvent(abortDeclineButton, 'press')

    expect(onGoBackTouched).toBeCalledTimes(1)
    expect(onDeclinedConformationTouched).toBeCalledTimes(0)
  })

  test('decline offer renders correctly', () => {
    const tree = render(
      <CommonDecline
        visible={true}
        declineType={DeclineType.CredentialOffer}
        didDeclineOfferOrProof={false}
        onGoBackTouched={onGoBackTouched}
        onDeclinedConformationTouched={onDeclinedConformationTouched}
      />
    )

    const confirmDeclineButton = tree.getByTestId(testIdWithKey('ConfirmDeclineButton'))
    const abortDeclineButton = tree.getByTestId(testIdWithKey('AbortDeclineButton'))

    expect(tree).toMatchSnapshot()
    expect(confirmDeclineButton).not.toBeNull()
    expect(abortDeclineButton).not.toBeNull()
  })
})
