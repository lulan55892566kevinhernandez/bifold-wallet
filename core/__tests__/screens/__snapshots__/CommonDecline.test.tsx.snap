// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common decline screen decline offer renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
      "bottom",
    ]
  }
>
  <RCTScrollView
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "height": "100%",
          "padding": 20,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#313132",
            "borderColor": "#FCBA19",
            "borderRadius": 5,
            "borderWidth": 1,
            "minWidth": 700,
            "padding": 10,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 5,
              "paddingTop": 5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "marginRight": 10,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FCBA19",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              î€‚
            </Text>
          </View>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#FFFFFF",
                "flexShrink": 1,
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
            testID="com.ariesbifold:id/HeaderText"
          >
            CredentialOffer.ConfirmDeclinedTitle
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "marginLeft": 40,
              "paddingBottom": 5,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "flexShrink": 1,
                "fontSize": 18,
                "fontWeight": "normal",
                "marginVertical": 16,
              }
            }
            testID="com.ariesbifold:id/BodyText"
          >
            CredentialOffer.ConfirmDeclinedMessage
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#781e4b",
                "borderRadius": 10,
              },
              Object {},
              Object {
                "elevation": 0,
              },
            ]
          }
        >
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            testID="com.ariesbifold:id/ShowCredentialDetails"
          >
            <View
              accessible={true}
              collapsable={false}
              nativeID="animatedComponent"
              style={
                Object {
                  "backgroundColor": "#781e4b",
                  "borderRadius": 10,
                  "opacity": 1,
                }
              }
            >
              <View
                testID="com.ariesbifold:id/CredentialCard"
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "minHeight": 247.5,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "rgba(0, 0, 0, 0.24)",
                          "borderBottomLeftRadius": 10,
                          "borderTopLeftRadius": 10,
                          "width": 90,
                        },
                        Object {
                          "backgroundColor": "rgba(0, 0, 0, 0.24)",
                        },
                      ]
                    }
                    testID="com.ariesbifold:id/CredentialCardSecondaryBody"
                  />
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#ffffff",
                          "borderRadius": 8,
                          "height": 90,
                          "justifyContent": "center",
                          "left": -52.5,
                          "top": 37.5,
                          "width": 90,
                        },
                        Object {
                          "elevation": 0,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#FFFFFF",
                            "fontSize": 18,
                            "fontWeight": "normal",
                          },
                          Object {
                            "alignSelf": "center",
                            "fontSize": 45,
                            "fontWeight": "bold",
                          },
                        ]
                      }
                    >
                      C
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "flexGrow": 1,
                        "padding": 37.5,
                      }
                    }
                    testID="com.ariesbifold:id/CredentialCardPrimaryBody"
                  >
                    <View
                      style={
                        Object {
                          "margin": -1,
                          "marginLeft": -52.5,
                        }
                      }
                    >
                      <View>
                        <View
                          style={
                            Object {
                              "flexDirection": "row",
                            }
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#FFFFFF",
                                  "fontSize": 14,
                                  "fontWeight": "bold",
                                },
                                Object {
                                  "color": "#FFFFFF",
                                  "flexShrink": 1,
                                },
                                Object {
                                  "flex": 1,
                                  "flexWrap": "wrap",
                                  "lineHeight": 19,
                                  "opacity": 0.8,
                                },
                              ]
                            }
                            testID="com.ariesbifold:id/CredentialIssuer"
                          >
                            d7e7f641-e0cc-46bc-908a-f351e2486b03
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "flexDirection": "row",
                            }
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#FFFFFF",
                                  "fontSize": 18,
                                  "fontWeight": "normal",
                                },
                                Object {
                                  "color": "#FFFFFF",
                                  "flexShrink": 1,
                                },
                                Object {
                                  "flex": 1,
                                  "flexWrap": "wrap",
                                  "fontWeight": "bold",
                                  "lineHeight": 24,
                                },
                              ]
                            }
                            testID="com.ariesbifold:id/CredentialName"
                          >
                            Credential
                          </Text>
                        </View>
                      </View>
                      <RCTScrollView
                        data={
                          Array [
                            undefined,
                            undefined,
                          ]
                        }
                        getItem={[Function]}
                        getItemCount={[Function]}
                        keyExtractor={[Function]}
                        onContentSizeChange={[Function]}
                        onLayout={[Function]}
                        onMomentumScrollBegin={[Function]}
                        onMomentumScrollEnd={[Function]}
                        onScroll={[Function]}
                        onScrollBeginDrag={[Function]}
                        onScrollEndDrag={[Function]}
                        removeClippedSubviews={false}
                        renderItem={[Function]}
                        scrollEnabled={false}
                        scrollEventThrottle={50}
                        stickyHeaderIndices={Array []}
                        viewabilityConfigCallbackPairs={Array []}
                      >
                        <View>
                          <View
                            onLayout={[Function]}
                            style={null}
                          />
                          <View
                            onLayout={[Function]}
                            style={null}
                          />
                        </View>
                      </RCTScrollView>
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "rgba(0, 0, 0, 0)",
                        "borderBottomLeftRadius": 10,
                        "borderTopRightRadius": 10,
                        "height": 90,
                        "justifyContent": "center",
                        "width": 90,
                      }
                    }
                    testID="com.ariesbifold:id/CredentialCardStatus"
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "rgba(0, 0, 0, 0)",
                            "borderBottomLeftRadius": 10,
                            "borderTopRightRadius": 10,
                            "height": 90,
                            "justifyContent": "center",
                            "width": 90,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </RNGestureHandlerButton>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "margin": 20,
        "marginTop": "auto",
      }
    }
  >
    <View
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      <View
        accessibilityLabel="CredentialOffer.ConfirmDecline"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#42803E",
            "borderRadius": 4,
            "opacity": 1,
            "padding": 16,
          }
        }
        testID="com.ariesbifold:id/ConfirmDeclineButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                false,
              ]
            }
          >
            CredentialOffer.ConfirmDecline
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingTop": 10,
          },
        ]
      }
    >
      <View
        accessibilityLabel="CredentialOffer.AbortDecline"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#42803E",
            "borderRadius": 4,
            "borderWidth": 2,
            "opacity": 1,
            "padding": 16,
          }
        }
        testID="com.ariesbifold:id/AbortDeclineButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#42803E",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                false,
              ]
            }
          >
            CredentialOffer.AbortDecline
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`common decline screen decline proof renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
      "bottom",
    ]
  }
>
  <RCTScrollView
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "height": "100%",
          "padding": 20,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#313132",
            "borderColor": "#FCBA19",
            "borderRadius": 5,
            "borderWidth": 1,
            "minWidth": 700,
            "padding": 10,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 5,
              "paddingTop": 5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "marginRight": 10,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FCBA19",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              î€‚
            </Text>
          </View>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#FFFFFF",
                "flexShrink": 1,
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
            testID="com.ariesbifold:id/HeaderText"
          >
            ProofRequest.ConfirmDeclinedTitle
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "marginLeft": 40,
              "paddingBottom": 5,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "flexShrink": 1,
                "fontSize": 18,
                "fontWeight": "normal",
                "marginVertical": 16,
              }
            }
            testID="com.ariesbifold:id/BodyText"
          >
            ProofRequest.ConfirmDeclinedMessage
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "margin": 20,
        "marginTop": "auto",
      }
    }
  >
    <View
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      <View
        accessibilityLabel="ProofRequest.ConfirmDecline"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#42803E",
            "borderRadius": 4,
            "opacity": 1,
            "padding": 16,
          }
        }
        testID="com.ariesbifold:id/ConfirmDeclineButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                false,
              ]
            }
          >
            ProofRequest.ConfirmDecline
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingTop": 10,
          },
        ]
      }
    >
      <View
        accessibilityLabel="ProofRequest.AbortDecline"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#42803E",
            "borderRadius": 4,
            "borderWidth": 2,
            "opacity": 1,
            "padding": 16,
          }
        }
        testID="com.ariesbifold:id/AbortDeclineButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#42803E",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                false,
              ]
            }
          >
            ProofRequest.AbortDecline
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`common decline screen did decline offer renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
      "bottom",
      "top",
    ]
  }
>
  <RCTScrollView
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "height": "100%",
          "padding": 20,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginTop": 25,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 26,
                "fontWeight": "bold",
              },
              Object {
                "fontWeight": "normal",
                "marginTop": 30,
                "textAlign": "center",
              },
            ]
          }
          testID="com.ariesbifold:id/RequestOrOfferDeclined"
        >
          CredentialOffer.CredentialDeclined
        </Text>
        <
          fill="#FFFFFF"
          height={250}
          style={
            Array [
              Object {
                "marginVertical": 66,
              },
            ]
          }
          width={250}
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "margin": 20,
        "marginTop": "auto",
      }
    }
  >
    <View
      accessibilityLabel="Global.Done"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#42803E",
          "borderRadius": 4,
          "opacity": 1,
          "padding": 16,
        }
      }
      testID="com.ariesbifold:id/Done"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              false,
            ]
          }
        >
          Global.Done
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`common decline screen did decline proof renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
      "bottom",
      "top",
    ]
  }
>
  <RCTScrollView
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "height": "100%",
          "padding": 20,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginTop": 25,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 26,
                "fontWeight": "bold",
              },
              Object {
                "fontWeight": "normal",
                "marginTop": 30,
                "textAlign": "center",
              },
            ]
          }
          testID="com.ariesbifold:id/RequestOrOfferDeclined"
        >
          ProofRequest.ProofRequestDeclined
        </Text>
        <
          fill="#FFFFFF"
          height={250}
          style={
            Array [
              Object {
                "marginVertical": 66,
              },
            ]
          }
          width={250}
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "margin": 20,
        "marginTop": "auto",
      }
    }
  >
    <View
      accessibilityLabel="Global.Done"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#42803E",
          "borderRadius": 4,
          "opacity": 1,
          "padding": 16,
        }
      }
      testID="com.ariesbifold:id/Done"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              false,
            ]
          }
        >
          Global.Done
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
